airbyte-commons/src/main/java/io/airbyte/commons/io/ArchiveHelper.java

```diff
@@ -0,0 +1,126 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.commons.io;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.zip.GZIPOutputStream;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArchiveHelper {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ArchiveHelper.class);
+
+  /**
+   * Compress a @param sourceFolder into a Gzip Tarball @param archiveFile
+   */
+  static public void createArchive(final Path sourceFolder, final Path archiveFile) throws IOException {
+    final TarArchiveOutputStream archive =
+        new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(archiveFile.toFile()))));
+    Files.walkFileTree(sourceFolder, new SimpleFileVisitor<>() {
+
+      @Override
+      public FileVisitResult visitFile(Path file, BasicFileAttributes attributes) {
+        // only copy files, no symbolic links
+        if (attributes.isSymbolicLink()) {
+          return FileVisitResult.CONTINUE;
+        }
+        Path targetFile = sourceFolder.relativize(file);
+        try {
+          compressFile(file, targetFile, archive);
+        } catch (IOException e) {
+          LOGGER.error(String.format("Failed to archive file %s: %s", file, e));
+          throw new RuntimeException(e);
+        }
+        return FileVisitResult.CONTINUE;
+      }
+
+      @Override
+      public FileVisitResult visitFileFailed(Path file, IOException exc) {
+        LOGGER.error(String.format("Failed to include file %s in archive", file));
+        return FileVisitResult.CONTINUE;
+      }
+
+    });
+    archive.close();
+  }
+
+  static private void compressFile(final Path file, final Path filename, final TarArchiveOutputStream archive) throws IOException {
+    final TarArchiveEntry tarEntry = new TarArchiveEntry(file.toFile(), filename.toString());
+    archive.putArchiveEntry(tarEntry);
+    Files.copy(file, archive);
+    archive.closeArchiveEntry();
+  }
+
+  /**
+   * Uncompress a Gzip Tarball @param archiveFile into the @param destinationFolder
+   */
+  static public void openArchive(final Path destinationFolder, final Path archiveFile) throws IOException {
```

Comment:
The ordering of the parameter isn't consistent with the create, should be archive first and destination then.

Could you rename the function? without looking at the comment I don't know what it does. what about `extractArchive`?